crates/sui-core/src/authority/authority_store.rs

```diff
@@ -326,29 +325,25 @@ impl AuthorityStore {
                             epoch_store.get_shared_locks(digest)?.into_iter().collect(),
                         )
                     })?;
-                    match shared_locks.get(id) {
-                        Some(version) => {
-                            if !self.object_version_exists(id, *version)? {
-                                // When this happens, other transactions that use smaller versions of
-                                // this shared object haven't finished execution.
-                                missing.push(ObjectKey(*id, *version));
-                            }
-                        }
-                        None => {
-                            // Abort the function because the lock should have been set.
-                            return Err(SuiError::SharedObjectLockNotSetError);
-                        }
+                    let Some(version) = shared_locks.get(id) else {
+                        // Abort the function because the lock should have been set.
+                        return Err(SuiError::SharedObjectLockNotSetError);
                     };
+                    if !self.object_version_exists(id, *version)? {
+                        // When this happens, other transactions that use smaller versions of this
+                        // shared object haven't finished execution.
+                        missing.push(MissingInput::from(ObjectKey(*id, *version)));
+                    }
                 }
                 InputObjectKind::MovePackage(id) => {
-                    if !self.object_version_exists(id, PACKAGE_VERSION)? {
-                        // The cert cannot have been formed if immutable inputs were missing.
-                        missing.push(ObjectKey(*id, PACKAGE_VERSION));
+                    if !self.object_exists(*id)? {
```

Comment:
How many versions can there be on the same object ID, for MovePackage?

Reply:
(Resolving as covered in the main comment thread).